dev/verify-db-queries.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+# This script runs all of our database queries (found in 
+# "/bluebutton-ansible-playbooks-data/dev/database-queries.txt") for 
+# a random number of beneficiaryIds.
+#
+# 	$ verify-db-queries.sh <database-name> <username> <number-of-randomb-benes>
+#
+
+DBNAME=$1
+USERNAME=$2
+NUMBENESTOQUERY=$3
+databaseQueries="/bluebutton-ansible-playbooks-data/dev/database-queries.txt"
+
+# select random sample of Beneficiaries.beneficiaryId
+randomBeneIds=( $(psql -U $USERNAME -d $DBNAME -c "SELECT beneficiaryId FROM Beneficiaries ORDER BY RANDOM() LIMIT $NUMBENESTOQUERY") )
+
+# set queries to timeout after 15 seconds
+setTimeout() {
+	$(psql -U $USERNAME -d $DBNAME -c "SET statement_timeout TO 15000")
+}
+
+# check if query exit status returned anything other than 0
+checkForTimeout() {
+	# FIXME exit status of 2 might be for timeouts - need to verify
+	if [ echo $? != 0]
+	then
+		$timedout=true
+	fi
+}
+
+runPsqlQuery() {
+	dbQuery=$1
+	psql -U $USERNAME -d $DBNAME $dbQuery

```

Comment:
This still won't work with the timeout: you're still running it in a separate process, which means it's a separate session. Try this, instead (per the `psql` man page: http://manpages.ubuntu.com/manpages/bionic/man1/psql.1.html):

```suggestion
	psql -U $USERNAME -d $DBNAME <<EOF
SET statement_timeout TO 15000;
${dbQuery}
EOF
```

This odd `<<EOF` construct is called a "heredoc" in Bash, and they're weird, but useful: http://tldp.org/LDP/abs/html/here-docs.html.

Reply:
Thanks for the documentation on this. I wasn't sure how to do it otherwise, obviously.